src/crypto/hmac_sha256.cpp

```diff
@@ -4,31 +4,32 @@
 
 #include <crypto/hmac_sha256.h>
 
-#include <string.h>
+#include <algorithm>
+#include <array>
+#include <cstring>
 
 CHMAC_SHA256::CHMAC_SHA256(const unsigned char* key, size_t keylen)
 {
-    unsigned char rkey[64];
-    if (keylen <= 64) {
-        memcpy(rkey, key, keylen);
-        memset(rkey + keylen, 0, 64 - keylen);
+    std::array<unsigned char, OUTPUT_SIZE * 2> rkey{};
+    if (keylen <= rkey.size()) {
+        std::memcpy(rkey.data(), key, keylen);
     } else {
-        CSHA256().Write(key, keylen).Finalize(rkey);
-        memset(rkey + 32, 0, 32);
+        CSHA256().Write(key, keylen).Finalize(rkey.data());
+        std::fill(rkey.begin() + OUTPUT_SIZE, rkey.end(), 0);
     }
 
-    for (int n = 0; n < 64; n++)
+    for (auto n = 0u; n < rkey.size(); n++)
```

Comment:
I'd prefer not to use 'auto' here, keeping integer signedness explicit makes it easier to reason about.